engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreActor.scala

```diff
@@ -1,37 +1,77 @@
 package cromwell.engine.workflow.workflowstore
 
 import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import akka.pattern.pipe
 import cats.data.NonEmptyList
 import cromwell.core.Dispatcher.EngineDispatcher
 import cromwell.core._
+import cromwell.services.metadata.MetadataService._
 import cromwell.util.GracefulShutdownHelper
 import cromwell.util.GracefulShutdownHelper.ShutdownCommand
 
+import scala.concurrent.Future
+
 final case class WorkflowStoreActor private(store: WorkflowStore, serviceRegistryActor: ActorRef)
   extends Actor with ActorLogging with GracefulShutdownHelper {
   import WorkflowStoreActor._
-
+  implicit val ec = context.dispatcher
+  
   lazy val workflowStoreSubmitActor: ActorRef = context.actorOf(WorkflowStoreSubmitActor.props(store, serviceRegistryActor), "WorkflowStoreSubmitActor")
   lazy val workflowStoreEngineActor: ActorRef = context.actorOf(
     WorkflowStoreEngineActor.props(store, serviceRegistryActor), "WorkflowStoreEngineActor")
 
   override def receive = {
     case ShutdownCommand => waitForActorsAndShutdown(NonEmptyList.of(workflowStoreSubmitActor))
+    case WorkflowStateRequest(id) => 
+      pipe(statusOf(id)).to(sender())
```

Comment:
`statusOf(id) pipeTo sender`?

Reply:
I can never get this `pipeTo` to compile but I'll try again

Reply:
I usually have that problem too, but I actually tried it and it worked. 🙂